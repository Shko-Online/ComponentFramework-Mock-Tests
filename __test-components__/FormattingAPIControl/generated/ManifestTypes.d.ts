/*
    Copyright (c) 2022 Betim Beja and Shko Online LLC
    Licensed under the MIT license.
*/

// Define IInputs and IOutputs Type. They should match with ControlManifest.
export interface IInputs {
    currencyInput: ComponentFramework.PropertyTypes.NumberProperty;
    dateInput: ComponentFramework.PropertyTypes.DateTimeProperty;
    decimalInput: ComponentFramework.PropertyTypes.DecimalNumberProperty;
    integerInput: ComponentFramework.PropertyTypes.WholeNumberProperty;
}
export interface IOutputs {
    currencyInput?: number;
    dateInput?: Date;
    decimalInput?: number;
    integerInput?: number;
}
