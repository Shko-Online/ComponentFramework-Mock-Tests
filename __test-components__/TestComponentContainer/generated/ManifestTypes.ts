/*
    Copyright (c) 2022 Betim Beja and Shko Online LLC
    Licensed under the MIT license.
*/

// Define IInputs and IOutputs Type. They should match with ControlManifest.
export interface IInputs {
    controlValue: ComponentFramework.PropertyTypes.StringProperty;
}
export interface IOutputs {
    controlValue?: string;
}
