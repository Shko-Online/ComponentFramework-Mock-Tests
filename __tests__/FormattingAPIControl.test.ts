/*
	Unless explicitly acquired and licensed from Licensor under another
	license, the contents of this file are subject to the Reciprocal Public
	License ("RPL") Version 1.5, or subsequent versions as allowed by the RPL,
	and You may not copy or use this file in either source code or executable
	form, except in compliance with the terms and conditions of the RPL.

	All software distributed under the RPL is provided strictly on an "AS
	IS" basis, WITHOUT WARRANTY OF ANY KIND, EITHER EXPRESS OR IMPLIED, AND
	LICENSOR HEREBY DISCLAIMS ALL SUCH WARRANTIES, INCLUDING WITHOUT
	LIMITATION, ANY WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR
	PURPOSE, QUIET ENJOYMENT, OR NON-INFRINGEMENT. See the RPL for specific
	language governing rights and limitations under the RPL. 
*/

import * as sinon from 'sinon';

import { ComponentFrameworkMockGenerator } from '@shko-online/componentframework-mock/ComponentFramework-Mock-Generator';
import { FormattingAPIControl }  from '@albanian-xrm/test-components/FormattingAPIControl';
import { IInputs, IOutputs }  from '@albanian-xrm/test-components/FormattingAPIControl/generated/ManifestTypes';
import { NumberPropertyMock } from '@shko-online/componentframework-mock/ComponentFramework-Mock/PropertyTypes/NumberProperty.mock';
import { DateTimePropertyMock } from '@shko-online/componentframework-mock/ComponentFramework-Mock/PropertyTypes/DateTimeProperty.mock';
import { DecimalNumberPropertyMock } from '@shko-online/componentframework-mock/ComponentFramework-Mock/PropertyTypes/DecimalNumberProperty.mock';
import { WholeNumberPropertyMock } from '@shko-online/componentframework-mock/ComponentFramework-Mock/PropertyTypes/WholeNumberProperty.mock';


describe("FormattingAPIControl tests", () => {
    let mockGenerator: ComponentFrameworkMockGenerator<IInputs,IOutputs>;
    beforeEach(()=>{
        const container = document.createElement('div');

         mockGenerator = new ComponentFrameworkMockGenerator(
            FormattingAPIControl,
            {
               currencyInput: NumberPropertyMock,
               dateInput: DateTimePropertyMock,
               decimalInput: DecimalNumberPropertyMock,
               integerInput: WholeNumberPropertyMock
            },
            container);
        document.body.appendChild(container);
    }
    )

    afterEach(()=>{
        document.body.innerHTML = null;
    })

    it("Init should work", () => {            
        mockGenerator.ExecuteInit();
        sinon.assert.calledOnce(mockGenerator.control.init);
        expect(document.body).toMatchSnapshot();
    })

    it("Update View should Work", () => {    
        mockGenerator.context.parameters.currencyInput.raw = 1000.01;
        mockGenerator.context.parameters.dateInput.raw = new Date(2022,8,2);
        mockGenerator.context.parameters.decimalInput.raw = 123.45;
        mockGenerator.context.parameters.integerInput.raw = 987;

        mockGenerator.ExecuteInit();
        mockGenerator.ExecuteUpdateView();
        sinon.assert.calledOnce(mockGenerator.control.init);
        sinon.assert.calledOnce(mockGenerator.control.updateView);
        expect(document.body).toMatchSnapshot();
    })
    it("Destroy should work", () => {
        mockGenerator.control.destroy();
		mockGenerator.ExecuteInit();
		mockGenerator.ExecuteUpdateView();
		expect(document.body).toMatchSnapshot(); 
	})
	it("getOuputs should work", () => {
        mockGenerator.control.getOutputs();
		mockGenerator.ExecuteInit();
		mockGenerator.ExecuteUpdateView();
		expect(document.body).toMatchSnapshot(); 
	})
})