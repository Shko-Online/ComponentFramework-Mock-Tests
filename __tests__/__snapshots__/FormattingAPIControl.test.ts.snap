// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormattingAPIControl tests Destroy should work 1`] = `
<body>
  <div>
    <div
      class="Formatting_Container"
    >
      <table
        class="FormattingControlSampleHtmlTable_HtmlTable"
      >
        <tr
          class="FormattingControlSampleHtmlTable_HtmlRow"
        >
          <th
            class="FormattingControlSampleHtmlTable_HtmlHeaderCell FormattingControlSampleHtmlTable_HtmlCell_Key"
          >
            Example Method
          </th>
          <th
            class="FormattingControlSampleHtmlTable_HtmlHeaderCell FormattingControlSampleHtmlTable_HtmlCell_Value"
          >
            Result
          </th>
        </tr>
        <tr
          class="FormattingControlSampleHtmlTable_HtmlRow"
        >
          <td
            class="FormattingControlSampleHtmlTable_HtmlCell FormattingControlSampleHtmlTable_HtmlCell_Key"
          >
            formatCurrency()
          </td>
          <td
            class="FormattingControlSampleHtmlTable_HtmlCell FormattingControlSampleHtmlTable_HtmlCell_Value"
          >
            $0.00
          </td>
        </tr>
        <tr
          class="FormattingControlSampleHtmlTable_HtmlRow"
        >
          <td
            class="FormattingControlSampleHtmlTable_HtmlCell FormattingControlSampleHtmlTable_HtmlCell_Key"
          >
            formatDecimal()
          </td>
          <td
            class="FormattingControlSampleHtmlTable_HtmlCell FormattingControlSampleHtmlTable_HtmlCell_Value"
          >
            0
          </td>
        </tr>
        <tr
          class="FormattingControlSampleHtmlTable_HtmlRow"
        >
          <td
            class="FormattingControlSampleHtmlTable_HtmlCell FormattingControlSampleHtmlTable_HtmlCell_Key"
          >
            formatInteger()
          </td>
          <td
            class="FormattingControlSampleHtmlTable_HtmlCell FormattingControlSampleHtmlTable_HtmlCell_Value"
          >
            0
          </td>
        </tr>
        <tr
          class="FormattingControlSampleHtmlTable_HtmlRow"
        >
          <td
            class="FormattingControlSampleHtmlTable_HtmlCell FormattingControlSampleHtmlTable_HtmlCell_Key"
          >
            formatLanguage()
          </td>
          <td
            class="FormattingControlSampleHtmlTable_HtmlCell FormattingControlSampleHtmlTable_HtmlCell_Value"
          >
            1033
          </td>
        </tr>
        <tr
          class="FormattingControlSampleHtmlTable_HtmlRow"
        >
          <td
            class="FormattingControlSampleHtmlTable_HtmlCell FormattingControlSampleHtmlTable_HtmlCell_Key"
          >
            formatDateYearMonth()
          </td>
          <td
            class="FormattingControlSampleHtmlTable_HtmlCell FormattingControlSampleHtmlTable_HtmlCell_Value"
          >
            September 2022
          </td>
        </tr>
        <tr
          class="FormattingControlSampleHtmlTable_HtmlRow"
        >
          <td
            class="FormattingControlSampleHtmlTable_HtmlCell FormattingControlSampleHtmlTable_HtmlCell_Key"
          >
            getWeekOfYear()
          </td>
          <td
            class="FormattingControlSampleHtmlTable_HtmlCell FormattingControlSampleHtmlTable_HtmlCell_Value"
          >
            37
          </td>
        </tr>
      </table>
    </div>
  </div>
</body>
`;

exports[`FormattingAPIControl tests Init should work 1`] = `
<body>
  <div>
    <div
      class="Formatting_Container"
    />
  </div>
</body>
`;

exports[`FormattingAPIControl tests Update View should Work 1`] = `
<body>
  <div>
    <div
      class="Formatting_Container"
    >
      <table
        class="FormattingControlSampleHtmlTable_HtmlTable"
      >
        <tr
          class="FormattingControlSampleHtmlTable_HtmlRow"
        >
          <th
            class="FormattingControlSampleHtmlTable_HtmlHeaderCell FormattingControlSampleHtmlTable_HtmlCell_Key"
          >
            Example Method
          </th>
          <th
            class="FormattingControlSampleHtmlTable_HtmlHeaderCell FormattingControlSampleHtmlTable_HtmlCell_Value"
          >
            Result
          </th>
        </tr>
        <tr
          class="FormattingControlSampleHtmlTable_HtmlRow"
        >
          <td
            class="FormattingControlSampleHtmlTable_HtmlCell FormattingControlSampleHtmlTable_HtmlCell_Key"
          >
            formatCurrency()
          </td>
          <td
            class="FormattingControlSampleHtmlTable_HtmlCell FormattingControlSampleHtmlTable_HtmlCell_Value"
          >
            $1,000.01
          </td>
        </tr>
        <tr
          class="FormattingControlSampleHtmlTable_HtmlRow"
        >
          <td
            class="FormattingControlSampleHtmlTable_HtmlCell FormattingControlSampleHtmlTable_HtmlCell_Key"
          >
            formatDecimal()
          </td>
          <td
            class="FormattingControlSampleHtmlTable_HtmlCell FormattingControlSampleHtmlTable_HtmlCell_Value"
          >
            123.45
          </td>
        </tr>
        <tr
          class="FormattingControlSampleHtmlTable_HtmlRow"
        >
          <td
            class="FormattingControlSampleHtmlTable_HtmlCell FormattingControlSampleHtmlTable_HtmlCell_Key"
          >
            formatInteger()
          </td>
          <td
            class="FormattingControlSampleHtmlTable_HtmlCell FormattingControlSampleHtmlTable_HtmlCell_Value"
          >
            987
          </td>
        </tr>
        <tr
          class="FormattingControlSampleHtmlTable_HtmlRow"
        >
          <td
            class="FormattingControlSampleHtmlTable_HtmlCell FormattingControlSampleHtmlTable_HtmlCell_Key"
          >
            formatLanguage()
          </td>
          <td
            class="FormattingControlSampleHtmlTable_HtmlCell FormattingControlSampleHtmlTable_HtmlCell_Value"
          >
            1033
          </td>
        </tr>
        <tr
          class="FormattingControlSampleHtmlTable_HtmlRow"
        >
          <td
            class="FormattingControlSampleHtmlTable_HtmlCell FormattingControlSampleHtmlTable_HtmlCell_Key"
          >
            formatDateYearMonth()
          </td>
          <td
            class="FormattingControlSampleHtmlTable_HtmlCell FormattingControlSampleHtmlTable_HtmlCell_Value"
          >
            September 2022
          </td>
        </tr>
        <tr
          class="FormattingControlSampleHtmlTable_HtmlRow"
        >
          <td
            class="FormattingControlSampleHtmlTable_HtmlCell FormattingControlSampleHtmlTable_HtmlCell_Key"
          >
            getWeekOfYear()
          </td>
          <td
            class="FormattingControlSampleHtmlTable_HtmlCell FormattingControlSampleHtmlTable_HtmlCell_Value"
          >
            35
          </td>
        </tr>
      </table>
    </div>
  </div>
</body>
`;

exports[`FormattingAPIControl tests getOuputs should work 1`] = `
<body>
  <div>
    <div
      class="Formatting_Container"
    >
      <table
        class="FormattingControlSampleHtmlTable_HtmlTable"
      >
        <tr
          class="FormattingControlSampleHtmlTable_HtmlRow"
        >
          <th
            class="FormattingControlSampleHtmlTable_HtmlHeaderCell FormattingControlSampleHtmlTable_HtmlCell_Key"
          >
            Example Method
          </th>
          <th
            class="FormattingControlSampleHtmlTable_HtmlHeaderCell FormattingControlSampleHtmlTable_HtmlCell_Value"
          >
            Result
          </th>
        </tr>
        <tr
          class="FormattingControlSampleHtmlTable_HtmlRow"
        >
          <td
            class="FormattingControlSampleHtmlTable_HtmlCell FormattingControlSampleHtmlTable_HtmlCell_Key"
          >
            formatCurrency()
          </td>
          <td
            class="FormattingControlSampleHtmlTable_HtmlCell FormattingControlSampleHtmlTable_HtmlCell_Value"
          >
            $0.00
          </td>
        </tr>
        <tr
          class="FormattingControlSampleHtmlTable_HtmlRow"
        >
          <td
            class="FormattingControlSampleHtmlTable_HtmlCell FormattingControlSampleHtmlTable_HtmlCell_Key"
          >
            formatDecimal()
          </td>
          <td
            class="FormattingControlSampleHtmlTable_HtmlCell FormattingControlSampleHtmlTable_HtmlCell_Value"
          >
            0
          </td>
        </tr>
        <tr
          class="FormattingControlSampleHtmlTable_HtmlRow"
        >
          <td
            class="FormattingControlSampleHtmlTable_HtmlCell FormattingControlSampleHtmlTable_HtmlCell_Key"
          >
            formatInteger()
          </td>
          <td
            class="FormattingControlSampleHtmlTable_HtmlCell FormattingControlSampleHtmlTable_HtmlCell_Value"
          >
            0
          </td>
        </tr>
        <tr
          class="FormattingControlSampleHtmlTable_HtmlRow"
        >
          <td
            class="FormattingControlSampleHtmlTable_HtmlCell FormattingControlSampleHtmlTable_HtmlCell_Key"
          >
            formatLanguage()
          </td>
          <td
            class="FormattingControlSampleHtmlTable_HtmlCell FormattingControlSampleHtmlTable_HtmlCell_Value"
          >
            1033
          </td>
        </tr>
        <tr
          class="FormattingControlSampleHtmlTable_HtmlRow"
        >
          <td
            class="FormattingControlSampleHtmlTable_HtmlCell FormattingControlSampleHtmlTable_HtmlCell_Key"
          >
            formatDateYearMonth()
          </td>
          <td
            class="FormattingControlSampleHtmlTable_HtmlCell FormattingControlSampleHtmlTable_HtmlCell_Value"
          >
            September 2022
          </td>
        </tr>
        <tr
          class="FormattingControlSampleHtmlTable_HtmlRow"
        >
          <td
            class="FormattingControlSampleHtmlTable_HtmlCell FormattingControlSampleHtmlTable_HtmlCell_Key"
          >
            getWeekOfYear()
          </td>
          <td
            class="FormattingControlSampleHtmlTable_HtmlCell FormattingControlSampleHtmlTable_HtmlCell_Value"
          >
            37
          </td>
        </tr>
      </table>
    </div>
  </div>
</body>
`;
